codalab/client/json_api_client.py

```diff
@@ -468,3 +470,61 @@ def update_authenticated_user(self, data, params=None):
                 path=self._get_resource_path('user'),
                 query_params=self._pack_params(params),
                 data=self._pack_document(data, 'users')))
+
+    @wrap_exception('Unable to fetch contents info of bundle {1}')
+    def fetch_contents_info(self, bundle_id, target_path='', depth=0):
+        request_path = '/bundles/%s/contents/info/%s' % \
+                       (bundle_id, urllib.quote(target_path))
+        response = self._make_request('GET', request_path,
+                                      query_params={'depth': depth})
+        return response['data']
+
+    @wrap_exception('Unable to fetch contents blob of bundle {1}')
+    def fetch_contents_blob(self, bundle_id, target_path='', range_=None,
+                            head=None, tail=None):
+        """
+        Returns a file-like object for the target on the given bundle.
+
+        :param bundle_id: id of target bundle
+        :param target_path: path to target in bundle
+        :param range_: range of bytes to fetch
+        :param head: number of lines to summarize from beginning of file
+        :param tail: number of lines to summarize from end of file
+        :return: file-like object containing requested data blob
+        """
+        request_path = '/bundles/%s/contents/blob/%s' % \
+                       (bundle_id, urllib.quote(target_path))
+        headers = {'Accept-Encoding': 'gzip'}
+        if range_ is not None:
+            headers['Range'] = 'bytes=%d-%d' % range_
+        params = {}
+        if head is not None:
+            params['head'] = head
+        if tail is not None:
+            params['tail'] = tail
+        response = self._make_request('GET', request_path, headers=headers,
+                                      query_params=params, return_response=True)
+
+        if response.headers.get('Content-Encoding') == 'gzip':
+            return un_gzip_stream(response)
+        return response
+
+    @wrap_exception('Unable to upload contents of bundle {1}')
+    def upload_contents_blob(self, bundle_id, fileobj=None, params=None,
+                             progress_callback=None):
+        request_path = '/bundles/%s/contents/blob/' % bundle_id
```

Comment:
Please document the arguments here, especially since the uploading logic has so many cases.